Makefile

```diff
@@ -28,33 +28,46 @@ compile:
 build-executable-script:
 	mkdir -p $(LBIN_DIR)
 	echo '#!/bin/bash' > $(LBIN_DIR)/$(EXECUTABLE_SCRIPT_NAME)
-	echo 'script_dir=$$(dirname "$$(readlink -f "$$0")")' >> $(LBIN_DIR)/$(EXECUTABLE_SCRIPT_NAME)
-	echo 'export PERL5LIB=$$script_dir/../$(LIB_DIR):$$PATH:$$PERL5LIB' >> $(LBIN_DIR)/$(EXECUTABLE_SCRIPT_NAME)
+	echo 'if [ -L $$0 ] ; then' >> $(LBIN_DIR)/$(EXECUTABLE_SCRIPT_NAME)
+	echo 'script_dir=$$(cd "$$(dirname "$$(readlink $$0)")"; pwd -P)' >> $(LBIN_DIR)/$(EXECUTABLE_SCRIPT_NAME)
+	echo 'else' >> $(LBIN_DIR)/$(EXECUTABLE_SCRIPT_NAME)
+	echo 'script_dir=$$(cd "$$(dirname "$$0")"; pwd -P)' >> $(LBIN_DIR)/$(EXECUTABLE_SCRIPT_NAME)
+	echo 'fi' >> $(LBIN_DIR)/$(EXECUTABLE_SCRIPT_NAME)
+	echo 'export PERL5LIB=$$script_dir/../$(LIB_DIR):$$PERL5LIB' >> $(LBIN_DIR)/$(EXECUTABLE_SCRIPT_NAME)
 	echo 'perl $$script_dir/../$(LIB_DIR)/$(SERVICE_CAPS)/$(SERVICE_CAPS)Server.pm $$1 $$2 $$3' >> $(LBIN_DIR)/$(EXECUTABLE_SCRIPT_NAME)
 	chmod +x $(LBIN_DIR)/$(EXECUTABLE_SCRIPT_NAME)
 
 build-startup-script:
 	mkdir -p $(LBIN_DIR)
 	echo '#!/bin/bash' > $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
-	echo 'script_dir=$$(dirname "$$(readlink -f "$$0")")' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
+	echo 'if [ -L $$0 ] ; then' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
+	echo 'script_dir=$$(cd "$$(dirname "$$(readlink $$0)")"; pwd -P)' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
+	echo 'else' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
+	echo 'script_dir=$$(cd "$$(dirname "$$0")"; pwd -P)' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
+	echo 'fi' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
 	echo 'export KB_DEPLOYMENT_CONFIG=$$script_dir/../deploy.cfg' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
-	echo 'export PERL5LIB=$$script_dir/../$(LIB_DIR):$$PATH:$$PERL5LIB' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
+	echo 'export PERL5LIB=$$script_dir/../$(LIB_DIR):$$PERL5LIB' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
 	echo 'plackup $$script_dir/../$(LIB_DIR)/$(SERVICE_CAPS).psgi' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
 	chmod +x $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
 
 build-test-script:
 	echo '#!/bin/bash' > $(TEST_DIR)/$(TEST_SCRIPT_NAME)
-	echo 'script_dir=$$(dirname "$$(readlink -f "$$0")")' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
-	echo 'export KB_DEPLOYMENT_CONFIG=$$script_dir/../deploy.cfg' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'echo "Running $$0 with args $$@"' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'if [ -L $$0 ] ; then' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'test_dir=$$(cd "$$(dirname "$$(readlink $$0)")"; pwd -P) # for symbolic link' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'else' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'test_dir=$$(cd "$$(dirname "$$0")"; pwd -P) # for normal file' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'fi' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
```

Comment:
prevent the error about `redline -f` not being a valid option from popping up on MacOS